fili-core/src/main/java/com/yahoo/bard/webservice/web/JsonApiResponseWriter.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+import com.yahoo.bard.webservice.data.Result;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * JsonApi writer for response.
+ */
+public class JsonApiResponseWriter implements ResponseWriter {
+
+    /**
+     * Writes JSON-API response.
+     * The response when serialized is in the following format:
+     * <pre>
+     * {@code{
+     *   "rows" : [
+     *     {
+     *       "dateTime" : "YYYY-MM-dd HH:mm:ss.SSS",
+     *       "logicalMetric1Name" : logicalMetric1Value,
+     *       "logicalMetric2Name" : logicalMetric2Value,
+     *       ...
+     *       "dimension1Name" : "dimension1ValueKeyValue1",
+     *       "dimension2Name" : "dimension2ValueKeyValue1",
+     *       ...
+     *     }, {
+     *      ...
+     *     }
+     *   ],
+     *   "dimension1Name" : [
+     *     {
+     *       "dimension1KeyFieldName" : "dimension1KeyFieldValue1",
+     *       "dimension1OtherFieldName" : "dimension1OtherFieldValue1"
+     *     }, {
+     *       "dimension1KeyFieldName" : "dimension1KeyFieldValue2",
+     *       "dimension1OtherFieldName" : "dimension1OtherFieldValue2"
+     *     },
+     *     ...
+     *   ],
+     *   "dimension2Name" : [
+     *     {
+     *       "dimension2KeyFieldName" : "dimension2KeyFieldValue1",
+     *       "dimension2OtherFieldName" : "dimension2OtherFieldValue1"
+     *     }, {
+     *       "dimension2KeyFieldName" : "dimension2KeyFieldValue2",
+     *       "dimension2OtherFieldName" : "dimension2OtherFieldValue2"
+     *     },
+     *     ...
+     *   ]
+     *   "linkName1" : "http://uri1",
+     *   "linkName2": "http://uri2",
+     *   ...
+     *   "linkNameN": "http://uriN"
+     * }
+     * }
+     * </pre>
+     *
+     * Where "linkName1" ... "linkNameN" are the N keys in paginationLinks, and "http://uri1" ... "http://uriN" are the
+     * associated URI's.
```

Comment:
I would push this description of the JSON format onto the class level description, since this is the entire purpose of the class.